src/Main.elm

```diff
@@ -52,26 +53,42 @@ update msg model =
     Decrement ->
       model - 1
 
+    Reset ->
+      init
+
 
 
 -- VIEW
 
 
 onIncrement: Attribute Msg
 onIncrement =
-  on "tap-increment" (Decode.succeed Increment)
+  on "tap-button" (Decode.succeed Increment)
 
 
 onDecrement: Attribute Msg
 onDecrement =
-  on "tap-decrement" (Decode.succeed Decrement)
+  on "tap-button" (Decode.succeed Decrement)
+
+onReset: Attribute Msg
+onReset =
+  on "tap-button" (Decode.succeed Reset)
 
 
 view : Model -> Html Msg
 view model =
-  node "x-yay"
-    [ attribute "count" (String.fromInt model)
-    , onIncrement
-    , onDecrement
+  let
+    isEven = (modBy 2 model) == 0
+
+    maybeResetButton =
+      if isEven then
+        node "x-button" [ attribute "text" "Reset", onReset ] []
+      else
+        node "noscript" [] []
+  in
+  node "x-app" []
```

Comment:
* Now the demo will add and remove a reset button based on the count. 
* This feature is not meant to be intuitive, rather it's meant to be a bit more difficult to render properly.
* During development I noticed a bug where elements were not being properly rendered after adding stateful ui with Elm. So I'd like to keep this functionality in order to continuously test this behavior.